src/Confluent.Kafka.Avro/SpecificSerializerImpl.cs

```diff
@@ -110,19 +110,19 @@ public byte[] Serialize(string topic, T data)
         {
             lock (serializeLockObj)
             {
-                if (!topicsRegistered.Contains(topic))
+                if (!subjectsRegistered.Contains(topic))
                 {
                     string subject = isKey
-                        ? schemaRegistryClient.ConstructKeySubjectName(topic)
-                        : schemaRegistryClient.ConstructValueSubjectName(topic);
+                        ? schemaRegistryClient.ConstructKeySubjectName(topic, typeof(T).FullName)
```

Comment:
check for instance of named schema and return fullname from the schema, not the class -- this assumes no case munging